src/common/util.cpp

```diff
@@ -1063,4 +1064,26 @@ std::string get_nix_version_display_string()
     strftime(buffer, sizeof(buffer), "%Y-%m-%d %H:%M:%S", &tm);
     return std::string(buffer);
   }
+
+  std::string get_human_readable_bytes(uint64_t bytes)
+  {
+    // Use 1024 for "kilo", 1024*1024 for "mega" and so on instead of the more modern and standard-conforming
+    // 1000, 1000*1000 and so on, to be consistent with other Monero code that also uses base 2 units
+    static const char *SIZES[] = { "B", "kB", "MB", "GB", "TB" };
+    if (bytes < 1024)
+    {
+      return boost::lexical_cast<std::string>(bytes) + " " + SIZES[0];
+    }
+    uint64_t div = 0;
+    uint64_t rem = 0;
+    while (bytes >= 1024 && div < 5)
+    {
+      rem = bytes % 1024;
+      div++;
+      bytes /= 1024;
+    }
+    double bytes_d = (double)bytes + (double)rem / 1024.0;
```

Comment:
Wow ... this leaves me somewhat speechless.

Well, if we ever need a *high speed* version of that method, and added complexity and probably also added temporary memory consumption no problem, I guess this could be it ...

Reply:
There is no temporary memory consumption. Everything is done in place. I've disassembled this, so I am very confident in that.

And what complexity? This is standard STL usage with a bit of math.

Reply:
Well then. I use your method now because it would be waste of your effort otherwise, and it does have a certain elegance.

But frankly, if you compare a single dead-simple loop with something that needs a `struct` definition, a static lookup table, another `struct` definition with one method, and some clever / non-trivial iteration over a map using all this, and then you ask "What [added] complexity?" somebody who does not eat, drink, and dream about STL probably won't understand you.

Reply:
If the struct + `upper_bound` was abhorrent, a loop over the struct elements would suffice. Logarithmic bounded lookups will not make much difference with a list this small, might even be a slight kill on perf. The bigger change is the pre-computation of the ranges.

And usage of `sort`, `binary_search`, `lower_bound`, and `upper_bound` in some ways seems fundamental to C++. I have found them to be useful tools across several contexts/problems.

> and some clever / non-trivial iteration over a map using all this, and then you ask "What [added] complexity?" somebody who does not eat, drink, and dream about STL probably won't understand you.

No map is ever created; a sorted array/vector is a binary tree with the root in the middle and its children split halfway, etc.